staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -255,17 +66,16 @@ func WithRequestManagementByClient(
 		}
 
 		for {
-			rmState := reqMgmt.curState.Load().(*RMState)
-			fs := reqMgmt.pickFlowSchema(r, rmState.flowSchemas, rmState.priorityLevelStates)
-			ps := reqMgmt.requestPriorityState(r, fs, rmState.priorityLevelStates)
-			if ps.config.Exempt {
+			rmState := reqMgmt.GetCurrentState()
+			fs := utilflowcontrol.PickFlowSchema(requestDigest, rmState.GetFlowSchemas(), rmState.GetPriorityLevelStates())
+			ps := utilflowcontrol.RequestPriorityState(requestDigest, fs, rmState.GetPriorityLevelStates())
```

Comment:
`PickFlowSchema` and `RequestPriorityState` can be combined into one function, which I think would naturally be called `Classify`.  Also, there are a lot of function calls to pick out arguments, from inside another package, which are then handed back to a function in that same package.  It would be more natural to wrap all those details inside `Classify` and make that a method of `rmState`.  Thus I suggest:
```
    fs, ps := rmState.Classify(requestDigest)
```